bin/stripdata.js

```diff
@@ -121,6 +123,21 @@ function splitDeviceId(deviceId) {
 	return retlist;
 }
 
+function removeIDsAndPayload(data) {
+	if (program.stripAll) {
+		if (data.payload)
+			delete data.payload;
+		if (data.id)
+			delete data.id;
+		if (data.guid)
+			delete data.guid;
+		if (data.bolus)
+			delete data.bolus;
+		if (data.suppressed)
+			delete data.suppressed;
```

Comment:
We don't want to delete all of `suppressed`, only hash or remove the items in `suppressed` that need to be changed. The `suppressed` key contains information that the current temp basal is suppressing (such as the normal scheduled rate).

Given the amount of checking, you might want a separate function called `stripBasalSuppressedInfo`, and call it in `performDataStripping`. You could also confirm that the data is for a basal in `stripBasalSuppressedInfo` by testing that `data.type === 'basal'`